cloudinit/distros/__init__.py

```diff
@@ -845,7 +848,13 @@ def _get_package_mirror_info(mirror_info, data_source=None,
         # ec2 availability zones are named cc-direction-[0-9][a-d] (us-east-1b)
         # the region is us-east-1. so region = az[0:-1]
         if _EC2_AZ_RE.match(data_source.availability_zone):
```

Comment:
I think maybe we can just drop this re.match and instead just specificall check data_source.platform_type == "ec2".

Reply:
But wouldn't that interfere with past release ? We would like to have past releases still being able to user ec2 mirrors, right ?

Reply:
You are right, darn backward compatibility.